crates/bevy_gizmos/src/arrows.rs

```diff
@@ -0,0 +1,106 @@
+//! Additional Gizmo Functions -- Arrows
+
+use crate::prelude::Gizmos;
+use bevy_math::{Quat, Vec2, Vec3};
+use bevy_render::color::Color;
+
+pub struct ArrowBuilder<'a, 's> {
+    gizmos: &'a mut Gizmos<'s>,
+    start: Vec3,
+    end: Vec3,
+    color: Color,
+    tip_length: f32,
+}
+
+/// A builder returned by [`Gizmos::arrow`] and [`Gizmos::arrow_2d`]
+impl ArrowBuilder<'_, '_> {
+    /// Change the length of the tips to be `length`.
+    /// The default tip length is [length of the arrow]/10, which is sufficient for most applications.
+    ///
+    /// # example
+    /// ```
+    /// # use bevy_gizmos::prelude::*;
+    /// # use bevy_render::prelude::*;
+    /// # use bevy_math::prelude::*;
+    /// fn system(mut gizmos: Gizmos) {
+    ///     gizmos.arrow(Vec3::ZERO, Vec3::ONE, Color::GREEN)
+    ///         .with_tip_length(3.);
+    /// }
+    /// # bevy_ecs::system::assert_is_system(system);
+    /// ```
+    pub fn with_tip_length(&mut self, length: f32) {
+        self.tip_length = length;
+    }
+}
+
+impl Drop for ArrowBuilder<'_, '_> {
+    /// actually draws the arrow, by drawing lines with the stored [`Gizmos`]
+    fn drop(&mut self) {
+        // draw the body of the arrow
+        self.gizmos.line(self.start, self.end, self.color);
+        // now the hard part is to draw the head in a sensible way
+        // put us in a coordinate system where the arrow is pointing towards +x and ends at the origin
+        // then transform it so we're at the end of the arrow facing the right direction
+        let pointing = (self.end - self.start).normalize();
+        let rotation = Quat::from_rotation_arc(Vec3::X, pointing);
+        let tips = [(1, 0), (0, 1), (-1, 0), (0, -1)]
+            .into_iter()
+            .map(|(y, z)| Vec3 {
+                x: -1.,
+                y: y as f32,
+                z: z as f32,
+            })
+            .map(|v| v * self.tip_length)
+            .map(|v| rotation.mul_vec3(v))
+            .map(|v| v + self.end);
```

Comment:
I've a functional background, and I like functional combinators more than anyone, but using a single `.map` would improve readability and performance, as the std [recommends avoiding long chains of `map` on static size arrays](https://doc.rust-lang.org/stable/std/primitive.array.html#note-on-performance-and-stack-usage)

```suggestion
let tips = [
    Vec3::new(-1., 1., 0.),
    Vec3::new(-1., 0., 1.),
    Vec3::new(-1., -1., 0.),
    Vec3::new(-1., 0., -1.),
];
let tips = tips.map(|v| rotation * (v * self.tip_length) + self.end);
```

Reply:
Honestly what I was actually trying to avoid was rustfmt taking up like 20 lines for the `tips` initialization when using `Vec3{ x: ..., y: ..., z: ...,}` and apparently i forgot Vec3::new existed